.github/workflows/ci-pullrequest-main.yml

```diff
@@ -79,12 +79,44 @@ jobs:
   PackageDeploy:
     runs-on: ubuntu-latest
     needs: BuildTest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
 
     steps:
     - uses: actions/checkout@v3
     
     - name: Docker Setup BuildX
       uses: docker/setup-buildx-action@v2
+
+    - name: Set image tag
+      run: |
+        echo "DOCKER_TAG_SHA=ghcr.io/buerokratt/mock-classifier:${GITHUB_SHA}" >> $GITHUB_ENV
+        echo "DOCKER_TAG_LATEST=ghcr.io/buerokratt/mock-classifier:latest" >> $GITHUB_ENV
     
     - name: Docker Build
-      run: docker image build src/MockClassifier.Api/
+      run: docker image build src/MockClassifier.Api/ --tag $DOCKER_TAG_SHA
+
+    - name: List Docker images
+      run: docker images
+
+    - name: Install cosign
+      uses: sigstore/cosign-installer@d6a3abf1bdea83574e28d40543793018b6035605
+      with:
+        cosign-release: 'v1.7.1'
+
+    - name: Log in to GitHub container registry
+      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
```

Comment:
@keijoraamat as part of the PR, can you add some guidance to the `readme.md` on how to setup the environment vars or at least cover which vars are needed?

Reply:
The GITHUB_TOKEN is a special env var. It's set by GitHub and one can control it through `permissions:` part of the job. Did you meant that or something else? It's just that for me, it's quite clear and I don't know what exactly to write down. I would love to explain though.

Reply:
Let's discuss it

Reply:
Having discussed with @keijoraamat, it seems that `secrets.GITHUB_TOKEN` is built into GitHub so no human intervention required.